lib/compilers/rust.ts

```diff
@@ -122,11 +122,15 @@ export class RustCompiler extends BaseCompiler {
         if (possibleEditions.length > 0) {
             let defaultEdition: undefined | string;
             if (!this.compiler.semver || this.isNightly()) {
-                defaultEdition = '2021';
+                defaultEdition = '2024';
```

Comment:
Just pointing out for other reviewers that this is not the default edition rustc uses,
but the latest stable one:

From rust 1.86.0:
```
--edition 2015|2018|2021|2024
                        Specify which edition of the compiler to use when
                        compiling code. The default is 2015 and the latest
                        stable edition is 2024.
```